File: frontend/src/api/quickEntry.ts
import { request } from './client';

/**
 * REQ-025 快速录入（API-017，Phase2A 最小版）。
 *
 * 后端实际契约（backend/api/quick_entry.py，Task A f771e02）：
 * - POST /api/quick-entry（capture）：按 mode 保留草稿 / 转新文档 / 追加到已有文档。
 * - DELETE /api/quick-entry/{id}（discard）：仅 status=draft 且属于当前用户的条目可丢弃。
 *
 * 注：后端实现含 `source` 字段，docs/07 §3.9 契约草案（line 258）暂未列出，
 * 由 Task C 文档回写统一；前端按后端实际契约实现。
 */

/** 录入 mode：draft 保留草稿 / create_document 转新文档 / append_document 追加到已有文档。 */
export type QuickEntryMode = 'draft' | 'create_document' | 'append_document';

/** 条目状态：draft 草稿 / converted 已转文档或追加 / discarded 已丢弃。 */
export type QuickEntryStatus = 'draft' | 'converted' | 'discarded';

/** POST /api/quick-entry 与 DELETE /api/quick-entry/{id} 返回的条目视图。 */
export type QuickEntryView = {
  id: number;
  status: QuickEntryStatus;
  created_document_id: number | null;
  target_document_id: number | null;
  title: string;
  owner_id: number;
};

export type QuickEntryCapturePayload = {
  title: string;
  content_md?: string;
  source?: string | null;
  target_document_id?: number | null;
  tag_ids?: number[];
  mode?: QuickEntryMode;
};

/** 录入一条快速条目（capture）：按 mode 保留草稿 / 转新文档 / 追加到已有文档。 */
export async function captureQuickEntry(
  token: string,
  payload: QuickEntryCapturePayload,
): Promise<QuickEntryView> {
  return request<QuickEntryView>('/api/quick-entry', {
    method: 'POST',
    token,
    body: JSON.stringify(payload),
  });
}

/** 丢弃一条 draft 条目（discard）：仅 status=draft 且属于当前用户的条目可丢弃。 */
export async function discardQuickEntry(token: string, entryId: number): Promise<QuickEntryView> {
  return request<QuickEntryView>(`/api/quick-entry/${entryId}`, {
    method: 'DELETE',
    token,
  });
}

File: backend/api/quick_entry.py
"""FastAPI router for REQ-025 quick entries (API-017, Phase2A minimal).

POST /api/quick-entry（capture：mode=draft / create_document / append_document）；
DELETE /api/quick-entry/{id}（discard：仅 status=draft → discarded）。
全部带空间隔离与文档权限过滤（service 层）。
"""

from __future__ import annotations

from backend.repository import repository
from backend.service.auth_context import TokenContext, get_current_user
from backend.service.quick_entry import (
    QuickEntryCaptureRequest,
    QuickEntryView,
    capture_quick_entry,
    discard_quick_entry,
)

try:
    from fastapi import APIRouter, Depends
    from pydantic import BaseModel
except ImportError:  # pragma: no cover - allows service tests before dependencies are installed
    APIRouter = None
    BaseModel = object


if APIRouter is not None:
    router = APIRouter(tags=["quick-entry"])

    class QuickEntryCaptureBody(BaseModel):
        title: str
        content_md: str = ""
        source: str | None = None
        target_document_id: int | None = None
        tag_ids: list[int] = []
        mode: str = "draft"

    @router.post("/api/quick-entry")
    def capture_quick_entry_endpoint(
        request: QuickEntryCaptureBody,
        ctx: TokenContext = Depends(get_current_user),
    ) -> dict[str, object]:
        view = capture_quick_entry(
            repository,
            ctx.user_id,
            ctx.current_space_id,
            QuickEntryCaptureRequest(
                title=request.title,
                content_md=request.content_md,
                source=request.source,
                target_document_id=request.target_document_id,
                tag_ids=tuple(request.tag_ids),
                mode=request.mode,
            ),
        )
        return {"code": 0, "msg": "ok", "data": _quick_entry_view(view)}

    @router.delete("/api/quick-entry/{entry_id}")
    def discard_quick_entry_endpoint(
        entry_id: int,
        ctx: TokenContext = Depends(get_current_user),
    ) -> dict[str, object]:
        view = discard_quick_entry(repository, ctx.user_id, ctx.current_space_id, entry_id)
        return {"code": 0, "msg": "ok", "data": _quick_entry_view(view)}

    def _quick_entry_view(view: QuickEntryView) -> dict[str, object]:
        return {
            "id": view.id,
            "status": view.status,
            "created_document_id": view.created_document_id,
            "target_document_id": view.target_document_id,
            "title": view.title,
            "owner_id": view.owner_id,
        }

else:
    router = None
